### PR TITLE
enforce min coverage %

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)

Describe the changes made in this PR.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [x] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/tests/yamato/check_coverage_percent.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def check_coverage(root_dir, min_percentage):
+    summary_xml = None
+    for dirpath, _, filenames in os.walk(root_dir):
+        if SUMMARY_XML_FILENAME in filenames:
+            summary_xml = os.path.join(dirpath, SUMMARY_XML_FILENAME)
+            break
+    if not summary_xml:
+        print("Couldn't find Summary.xml in root directory")
```
Comment (vincentpierre) [inline ml-agents/tests/yamato/check_coverage_percent.py]:
```suggestion
        print("Couldn't find "+ SUMMARY_XML_FILENAME +" in root directory")
```

File: ml-agents/tests/yamato/check_coverage_percent.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        sys.exit(0)
```
Comment (vincentpierre) [inline ml-agents/tests/yamato/check_coverage_percent.py]: I am not aware. What generates "Summary.xml"? In what circumstances is it not generated? Should this be an error code 1 if the file was not present?

File: ml-agents/tests/yamato/check_coverage_percent.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    with open(summary_xml) as f:
+        # Look for a line of the form
+        # <Linecoverage>73.9</Linecoverage>
+        lines = f.readlines()
+        for l in lines:
+            if "Linecoverage" in l:
+                pct = l.replace("<Linecoverage>", "").replace("</Linecoverage>", "")
```
Comment (vincentpierre) [inline ml-agents/tests/yamato/check_coverage_percent.py]: Should these tags or the string `Linecoverage` be static strings or are they guaranteed not to change and are okay yo hard code?

File: ml-agents/tests/yamato/check_coverage_percent.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                else:
+                    print(
+                        "Coverage {} is above the min percentage of {}.".format(
+                            pct, min_percentage
+                        )
+                    )
+                    sys.exit(0)
+
+    # Couldn't parse the file
+    print("Couldn't find Linecoverage in summary file")
```
Comment (vincentpierre) [inline ml-agents/tests/yamato/check_coverage_percent.py]:
```suggestion
    print("Couldn't find Linecoverage in summary file")
    sys.exit(1)
```
?

Comment (vincentpierre) [reply ml-agents/tests/yamato/check_coverage_percent.py]: If `Linecoverage` was not found, I think it should raise. And even if it should not raise, we should call `sys.exit(0)` for consistency.

File: ml-agents/tests/yamato/check_coverage_percent.py (hunk 2 above)
Comment (chriselion) [reply ml-agents/tests/yamato/check_coverage_percent.py]: It's an artifact of code coverage (which should always be in CI as of https://github.com/Unity-Technologies/ml-agents/pull/3633). 

Good point, that was a typo and it should be exit(1)

File: ml-agents/tests/yamato/check_coverage_percent.py (hunk 3 above)
Comment (chriselion) [reply ml-agents/tests/yamato/check_coverage_percent.py]: I don't know about "guaranteed not to change" but they're not expected not to change very often. I'd add some more comments that this is quick-and-dirty

File: ml-agents/tests/yamato/check_coverage_percent.py (hunk 1 above)
Comment (chriselion) [reply ml-agents/tests/yamato/check_coverage_percent.py]: Fixed

File: ml-agents/tests/yamato/check_coverage_percent.py (hunk 4 above)
Comment (chriselion) [reply ml-agents/tests/yamato/check_coverage_percent.py]: Fixed - the only exit(0) is if we find `Linecoverage` and it's over the threshold

**Comment by chriselion** [comment]
```
################################### Running next command ###################################
 python ml-agents/tests/yamato/check_coverage_percent.py upm-ci~/test-results/ 71 
############################################################################################
[21:18:51.073 INF] Coverage 73.9 is above the min percentage of 71.0.
```

**Comment by chriselion** [comment]
I'm currently expecting the most recent commit to fail - I set the coverage % to 80 to make sure that the exit(1) actually causes yamato to fail

**Pull request merged**